internal/refactoring/move_statement.go

```diff
@@ -100,63 +109,123 @@ func impliedMoveStatements(cfg *configs.Config, prevRunState *states.State, expl
 		// instance where the configuration _doesn't_ have count set.
 		// If so, we'll generate a statement replacing no-key with zero-key or
 		// vice-versa.
-		for _, rState := range modState.Resources {
-			rAddr := rState.Addr
-			rCfg := cfg.Module.ResourceByAddr(rAddr.Resource)
-			if rCfg == nil {
-				// If there's no configuration at all then there can't be any
-				// automatic move fixup to do.
-				continue
-			}
-			approxSrcRange := tfdiags.SourceRangeFromHCL(rCfg.DeclRange)
-
-			// NOTE: We're intentionally not checking to see whether the
-			// "to" addresses in our implied statements already have
-			// instances recorded in state, because ApplyMoves should
-			// deal with such conflicts in a deterministic way for both
-			// explicit and implicit moves, and we'd rather have that
-			// handled all in one place.
-
-			var fromKey, toKey addrs.InstanceKey
-
-			switch {
-			case rCfg.Count != nil:
-				// If we have a count expression then we'll use _that_ as
-				// a slightly-more-precise approximate source range.
-				approxSrcRange = tfdiags.SourceRangeFromHCL(rCfg.Count.Range())
-
-				if riState := rState.Instances[addrs.NoKey]; riState != nil {
-					fromKey = addrs.NoKey
-					toKey = addrs.IntKey(0)
-				}
-			case rCfg.Count == nil && rCfg.ForEach == nil: // no repetition at all
-				if riState := rState.Instances[addrs.IntKey(0)]; riState != nil {
-					fromKey = addrs.IntKey(0)
-					toKey = addrs.NoKey
-				}
-			}
+		into = append(into, impliedMoveStatementsForModuleResources(cfg, modState, explicitStmts)...)
+	}
 
-			if fromKey != toKey {
-				// We mustn't generate an implied statement if the user already
-				// wrote an explicit statement referring to this resource,
-				// because they may wish to select an instance key other than
-				// zero as the one to retain.
-				if !haveMoveStatementForResource(rAddr, explicitStmts) {
-					into = append(into, MoveStatement{
-						From:      addrs.ImpliedMoveStatementEndpoint(rAddr.Instance(fromKey), approxSrcRange),
-						To:        addrs.ImpliedMoveStatementEndpoint(rAddr.Instance(toKey), approxSrcRange),
-						DeclRange: approxSrcRange,
-						Implied:   true,
-					})
-				}
-			}
+	for _, childCfg := range cfg.Children {
+		into = append(into, impliedMoveStatements(childCfg, prevRunState, explicitStmts)...)
+	}
+
+	return into
+}
+
+// impliedMoveStatementsForModules creates implied move statements for module calls.
+func impliedMoveStatementsForModules(
+	prevRunState *states.State,
+	parentCfg *configs.Config,
+	modCallName string,
+	modCallCfg *configs.ModuleCall,
+	explicitStmts []MoveStatement) []MoveStatement {
+	var into []MoveStatement
+	var fromKey, toKey addrs.InstanceKey
+
+	parentState := prevRunState.Module(addrs.RootModuleInstance)
```

Comment:
I'm curious how this works for module calls a few levels deep within the configuration?  I would expect to see a addrs.Module address used here?  Perhaps parentCfg.Path?

I don't see a test for this scenario, could ensure there is one?